### DISCUSSION TITLE
How can I programmatically determine the Blazor render mode

### DISCUSSION THREAD

**Discussion by DavidThielen** [dotnet/aspnetcore#54582] [Q&A]
Hi all;

How can I get the render mode Blazor is using for my component. [There is this suggestion](https://stackoverflow.com/questions/77558237/determining-the-current-render-mode-in-blazor-when-set-to-auto-at-runtime) but that won't work for Auto as it could be either. I want to verify I've configured everything properly so my app is tunning in InteractiveServer.

thanks - dave

**Comment by danroth27** [comment]
Hi @DavidThielen. We have in our roadmap for .NET 9 to introduce an API for determining the current render mode: #49401. With .NET 8 you'll need to use a combination of check to deduce the render mode. See the comments on #49401 for some approaches that the Blazor community is currently using.

**Comment by mrpmorris** [comment]
See if this helps.

https://github.com/TimeWarpEngineering/blazor-state/blob/9b15f5d430f995092e0dc76d9f432630f5121ebf/Source/BlazorState/Components/BlazorStateComponent.cs#L14
